### PR TITLE
Update `PyInstaller.building.api.EXE.__init__`

### LINKED ISSUE #10844: `PyInstaller.building.api.EXE` is missing `hide_console`
Compare https://github.com/pyinstaller/pyinstaller/pull/7735 and

> (Windows) Add an option to hide or minimize the console window in console-enabled applications, but only if the program’s process owns the console window (i.e., the program was not launched from an existing console window). ([#7729](https://github.com/pyinstaller/pyinstaller/issues/7729))

from https://pyinstaller.org/en/v6.0.0/CHANGES.html#id1

### PR DESCRIPTION
**Pull request by Avasam**
Closes #10844

### CODE REVIEW

File: stubs/pyinstaller/PyInstaller/building/api.pyi [hunk 1]
```diff
@@ -118,17 +123,19 @@ class EXE(Target):
...
         console: bool = True,
+        hide_console: _HideConsole = None,
         disable_windowed_traceback: bool = False,
         debug: bool = False,
         name: str | None = None,
         icon: _IconParam = None,
         version: _VersionParam = None,
         manifest: _ManifestParam = None,
-        embed_manifest: bool = True,
+        embed_manifest: Literal[True] = True,
```
Comment (Avasam) [inline stubs/pyinstaller/PyInstaller/building/api.pyi:133]: Deprecated in 6.0, will raise with a non-truthy value

File: stubs/pyinstaller/PyInstaller/building/api.pyi [hunk 2]
```diff
@@ -118,17 +123,19 @@ class EXE(Target):
         exclude_binaries: bool = False,
         bootloader_ignore_signals: bool = False,
         console: bool = True,
+        hide_console: _HideConsole = None,
```
Comment (Avasam) [inline stubs/pyinstaller/PyInstaller/building/api.pyi:126]: Added in 6.0, missing from changelog

File: stubs/pyinstaller/PyInstaller/building/api.pyi [hunk 3]
```diff
@@ -118,17 +123,19 @@ class EXE(Target):
...
         icon: _IconParam = None,
         version: _VersionParam = None,
         manifest: _ManifestParam = None,
-        embed_manifest: bool = True,
+        embed_manifest: Literal[True] = True,
         resources: Sequence[str] = ...,
         strip: bool = False,
         upx_exclude: Iterable[str] = ...,
         runtime_tmpdir: str | None = None,
+        contents_directory: str = "_internal",
```
Comment (Avasam) [inline stubs/pyinstaller/PyInstaller/building/api.pyi:138]: Added in 6.1

File: stubs/pyinstaller/PyInstaller/building/api.pyi (hunk 1 above)
Comment (AlexWaygood) [reply stubs/pyinstaller/PyInstaller/building/api.pyi:133]: https://github.com/pyinstaller/pyinstaller/blob/dd40fb2cd75b86b5bbe23843bafde09740f451af/PyInstaller/building/api.py#L480-L484 reminds me very strongly of https://quotepark.com/quotes/1458843-douglas-adams-arthur-dent-what-happens-if-i-press-this-button/...

**Review by AlexWaygood — APPROVED** [review]

**Pull request merged**